### PR TITLE
[FLOSS H2] docs: Improve docs/COMPILE.md with clearer instructions

### PR DESCRIPTION
**Pull request by sedadas**
## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by sedadas** [comment]
@markus2330 Another thing that I noticed was that the compilation docs were a little bit confused about the EL versions (for example, the installation of the base libraries was done using `dnf` whereas everything else was done using `yum`; both are supported on EL 8.x, but only `yum` is available on EL 7.x). Additionally, the "Troubleshooting" section also only referred to issues on EL 7.x, whereas there is no mention if such issues occur on EL 8.x. This is perhaps a good first issue.

**Comment by markus2330** [comment]
> @markus2330 Another thing that I noticed was that the compilation docs were a little bit confused about the EL versions (for example, the installation of the base libraries was done using dnf whereas everything else was done using yum; both are supported on EL 8.x, but only yum is available on EL 7.x). Additionally, the "Troubleshooting" section also only referred to issues on EL 7.x, whereas there is no mention if such issues occur on EL 8.x. This is perhaps a good first issue.

Thank you for reporting, they are nice catches. They are not ideal for "good first issue" (at least not for everyone) as specific knowledge about EL is needed. So probably best is if you directly fix them as a H2 task. (Issue is not really needed, as it is  simply about updating/fixing information.)

**Comment by lawli3t** [comment]
Good fixes

Maybe we should change `apt-get` usages to `apt` while we're at it? Afaic it seems to be the preferred way of using apt now.

**Review by muskater — COMMENTED** [review]
This is a great addition to the compile guide. 
I just want to point out, that on Fedora `yum` was replaced with `dnf` a long time ago (see: https://fedoraproject.org/wiki/Yum). So we might want to consider having a separate section for Fedora. It is not absolutely necessary, since `yum` still works on newer Fedora installations.

**Comment by sedadas** [comment]
Thank you for the catches!
* About `dnf` and `yum` - they are both supported on newer releases of EL, so I'd leave the docs at `yum` to make it backwards compatible
* About `apt` and `apt-get` - I'd leave the docs with `apt-get` because of the same reasons
* About the last point with EL 8, I will spin up a VM these days to check if I can reproduce this

**Review by Nirnberger — APPROVED** [review]
Very good and detailed changes. There is nothing more to add, other than "great job" :+1:

File: doc/COMPILE.md [hunk 1]
```diff
@@ -52,56 +54,65 @@ installation of self-compiled Elektra (such as how to uninstall it).
...
 > The core of Elektra never depends on other libraries.
 
-To build documentation you need doxygen (we recommend 1.8.8+), graphviz and [ronn-ng](https://github.com/apjanke/ronn-ng/blob/master/INSTALLING.md):
+### Documentation dependencies
 
-```sh
-apt-get install doxygen graphviz
-gem install ronn-ng -v 0.10.1.pre1
-```
+To build documentation you need doxygen (we recommend 1.8.8+), graphviz and [ronn-ng](https://github.com/apjanke/ronn-ng/blob/master/INSTALLING.md). These can be installed as follows:
```
Comment (Nirnberger) [inline doc/COMPILE.md:59]: Useful changes and clarifications :+1:. It seems like a determiner slipped your attention here. 
```suggestion
To build the documentation you need doxygen (we recommend 1.8.8+), graphviz and [ronn-ng](https://github.com/apjanke/ronn-ng/blob/master/INSTALLING.md). These can be installed as follows:
```

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
Thank you, great job! :fireworks:

**Pull request merged**